generators/server/templates/src/test/java/package/config/TestLazyBeanInitConfiguration.java.ejs

```diff
@@ -0,0 +1,49 @@
+<%#
+ Copyright 2013-2019 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://jhipster.github.io/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%=packageName%>.config;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+
+/**
+ * This class allows you to run your tests without initializing unneeded beans.
+ * <br/>
+ * In order to override this for any specific test, add to it: {@code @ActiveProfiles(TestLazyBeanInitConfiguration.EAGER_BEAN_INIT)}
+ * <br/>
+ * When working with spring-boot 2.2.0 or later, should be replaced with property: <code>spring.main.lazy-initialization</code>.
+ * <br/>
+ * See <a href="https://spring.io/blog/2019/03/14/lazy-initialization-in-spring-boot-2-2">lazy-initialization-in-spring-boot-2-2</a>.
+ */
+@Component
+@Profile("!" + TestLazyBeanInitConfiguration.EAGER_BEAN_INIT)
+public class TestLazyBeanInitConfiguration {
```

Comment:
shouldn't you need to implement `BeanFactoryPostProcessor`?

Reply:
you are completely right, probably got lost on the way to the PR, I'll return it